src/AnchoredOverlay/AnchoredOverlay.tsx

```diff
@@ -154,7 +161,6 @@ export const AnchoredOverlay: React.FC<AnchoredOverlayProps> = ({
           onEscape={onEscape}
           ref={updateOverlayRef}
           role="listbox"
-          visibility={position ? 'visible' : 'hidden'}
```

Comment:
`visibility='hidden'` [removes an element from the Accessibility Tree](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#accessibility_concerns), so I was initially concerned that removing the `visibility` prop would cause a hidden `Overlay` to remain in the AT. 

However, IIUC, this PR dispenses with the concept of “a hidden `Overlay`” (instead [rendering `null`](https://github.com/primer/components/blob/8374dd5b5995a73beab9b953a54b8eb25b292322/src/AnchoredOverlay/AnchoredOverlay.tsx#L171)).